modin/backends/pandas/query_compiler.py

```diff
@@ -630,6 +630,197 @@ def sort_index_for_equal_values(result, ascending):
 
     # END Reduction operations
 
+    def _resample_func(
+        self, resample_args, func_name, new_columns=None, df_op=None, *args, **kwargs
+    ):
+        def map_func(df):
+            if df_op is not None:
+                df = df_op(df)
+            resampled_val = df.resample(*resample_args)
+            op = getattr(pandas.core.resample.Resampler, func_name)
+            if callable(op):
+                try:
+                    # This will happen with Arrow buffer read-only errors. We don't want to copy
+                    # all the time, so this will try to fast-path the code first.
+                    val = op(resampled_val, *args, **kwargs)
+                except (ValueError):
+                    resampled_val = df.copy().resample(*resample_args)
+                    val = op(resampled_val, *args, **kwargs)
+            else:
+                val = getattr(resampled_val, func_name)
+
+            if isinstance(val, pandas.Series):
+                return val.to_frame()
+            else:
+                return val
+
+        new_modin_frame = self._modin_frame._apply_full_axis(
+            axis=0, func=map_func, new_columns=new_columns
+        )
+        return self.__constructor__(new_modin_frame)
+
+    def res_get_gruop(self, resample_args, name, obj):
```

Comment:
Also looking through the documentation, many of these can be rewritten as `groupby`. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.resample.Resampler.median.html

We could add those in the API layer (`dataframe.py`). The others can be here in the query compiler. Does this make sense?

Reply:
Do you mean use `groupby` functionality directly in API layer instead of using additional function in `query_compiler` where it is possible for `Resampler`? Ok, I implement this.

Reply:
@devin-petersohn I investigate possibilities of reusing `groupby` functionality. But I don't understand how to use `resample` parameters in this case and connect this with `groupby` call. Can you describe me your idea?